cli/main.c

```diff
@@ -157,33 +144,65 @@ int main(int argc, char** argv) {
   // FIXME: this is temp till arg parse implemented.
   PkResult result;
 
-  if (argc == 1) {
+  static const char* const usage[] = {
+    "pocket [option] ... [-c cmd | file | -] [arg] ...",
+    NULL,
+  };
+
+  int debug = 0;
+  int quiet = 0;
+  int version = 0;
+  const char* cmd = NULL;
+
+  struct argparse_option flags[] = {
+    OPT_HELP(),
+    OPT_GROUP("Basic options"),
+    OPT_STRING('c', "cmd", &cmd, "Evaluate and run the passed string."),
+    OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version."),
+    OPT_BOOLEAN('q', "quiet", &quiet,
+               "Don't print version and copyright statement on REPL startup."),
+    OPT_BOOLEAN('v', "version", &version,
+               "Prints the pocketlang version number and exit."),
+    OPT_END(),
+  };
+
+  struct argparse argparse;
+  argparse_init(&argparse, flags, usage, 0);
+  argparse_describe(&argparse,
+        "\nPocketlang is a small and fast functional language written in C.",
+        "\nCopyright(c) 2020 - 2021 ThakeeNathees." \
+        "\nFree and open source software under the terms of the MIT license.");
+  argc = argparse_parse(&argparse, argc, argv);
```

Comment:
Argparsing should be at the top of the main, just move all these above.

Reply:
Hm! I see. But have a look at the current PR, we can't just move it at the top. The temporary arg parsing and the library one has been now merged. I just have a problem running `./pocket <file.pk>` now. Thanks!

Reply:
I don't get it, why is it impossible to move to the top of the main. (like this)
```c
int main(int argc, const char** argv) {
  // All the parsing here.

  // Rest of the code here.
}
```

Reply:
For example, `options` is part of the parsing code, do you see what I mean?

Reply:
No, you shouldn't use options when parsing, Just get the values from argparse and update using the values once the parsing is done.
1. Get the command line options
2. update the `options` variable
3. rest of the code.

```c
  bool debug;
  OPT_BOOLEAN('d', "debug", &debug, "Compile and run a debug version.")

  optinos.debug = debug;
  ...
```

Reply:
Okay, I'll do that. I also realized the argparse library doesn't follow the conventional argc & argv values. `./pocket` means argc is 0 and and argv[0] is null.

Just to note this while rearranging the code to suit the specification you've mentioned above. Thanks!